src/pal/src/misc/environ.cpp

```diff
@@ -634,11 +661,422 @@ SetEnvironmentVariableA(
+            [in] If this is TRUE, "name=" will unset the 'name' variable.
+
+Return Values
+
+    TRUE on success, FALSE otherwise
+
+--*/
+BOOL EnvironPutenv(const char* entry, BOOL deleteIfEmpty)
+{
+    BOOL result = FALSE;
+
+    bool fOwningCS = false;
+
+    CPalThread * pthrCurrent = InternalGetCurrentThread();
+
+    const char *equalsSignPosition = strchr(entry, '=');
+    if (equalsSignPosition == entry || equalsSignPosition == nullptr)
+    {
+        // "=foo" and "foo" have no meaning
+        return FALSE;
+    }
+
+    char* copy = strdup(entry);
+    if (copy == nullptr)
+    {
+        return FALSE;
+    }
+
+    int nameLength = equalsSignPosition - entry;
+
+    if (equalsSignPosition[1] == '\0' && deleteIfEmpty)
+    {
+        // "foo=" removes foo from the environment in _putenv() on Windows.
+        // The same string can result from a call to SetEnvironmentVariable()
+        // with the empty string as the value, but in that case we want to
+        // set the variable's value to "". deleteIfEmpty will be FALSE in
+        // that case.
+
+        // Change '=' to '\0'
+        copy[nameLength] = '\0';
+
+        EnvironUnsetenv(copy);
+        InternalFree(copy);
+
+        result = TRUE;
+    }
+    else
+    {
+        // See if we are replacing an item or adding one.
+
+        InternalEnterCriticalSection(pthrCurrent, &gcsEnvironment);
+        fOwningCS = true;
+
+        int i;
+        for (i = 0; palEnvironment[i] != nullptr; i++)
+        {
+            const char *existingEquals = strchr(palEnvironment[i], '=');
+            if (existingEquals == nullptr)
+            {
+                // The PAL screens out malformed strings, but the strings which
+                // came from the system during initialization might not have the
+                // equals sign. We treat the entire string as a name in that case.
+                existingEquals = palEnvironment[i] + strlen(palEnvironment[i]);
+            }
+
+            if (existingEquals - palEnvironment[i] == nameLength)
+            {
+                if (memcmp(entry, palEnvironment[i], nameLength) == 0)
+                {
+                    InternalFree(palEnvironment[i]);
+                    palEnvironment[i] = copy;
+
+                    result = TRUE;
+                    break;
+                }
+            }
+        }
+
+        if (palEnvironment[i] == nullptr)
+        {
+            _ASSERTE(i <= palEnvironmentCapacity);
+            if (i == palEnvironmentCapacity)
+            {
+                // We need more space in our environment, so let's double the size.
+                int resizeRet = ResizeEnvironment(palEnvironmentCapacity * 2);
+                if (resizeRet != TRUE)
+                {
+                    InternalFree(copy);
+                    goto done;
+                }
+            }
+
+            _ASSERTE(copy != nullptr);
+            palEnvironment[i] = copy;
+            palEnvironment[i + 1] = nullptr;
```

Comment:
This has been addressed, but I'm not sure why GitHub is showing the prior diff in the preview. Please view the full file to see the latest version.